https/callBuilder.go

```diff
@@ -662,10 +662,11 @@ func (cb *defaultCallBuilder) CallAsJson() (*json.Decoder, *http.Response, error
 
 	if result.Response != nil {
 		if result.Response.Body == http.NoBody {
-			err = fmt.Errorf("response body empty")
+			return nil, result.Response, fmt.Errorf("response body empty")
 		}
-
-		return json.NewDecoder(result.Response.Body), result.Response, err
+		bodyBytes, err := io.ReadAll(result.Response.Body)
+		result.Response.Body = io.NopCloser(bytes.NewBuffer(bodyBytes))
+		return json.NewDecoder(io.NopCloser(bytes.NewBuffer(bodyBytes))), result.Response, err
```

Comment:
What is the purpose of these lines of code? Also, why is `io.NopCloser(bytes.NewBuffer(bodyBytes))` used twice?

Reply:
the purpose of this line is to create a `io.ReadCloser` that is used twice. One is the initialize the Respons.Body and other to initialize Decoder. We cannot create them at one line as this `io.NopCloser` func returns the address of created object which can be updated by any of `decoder` OR `Response.Body`.